src/utils/Processing.ts

```diff
@@ -99,7 +99,22 @@ export function loadExternalDependencies(
 }
 
 function getLakeOfFHIR(inDir: string): LakeOfFHIR {
-  const files = getFilesRecursive(inDir).filter(file => file.endsWith('.json'));
+  // Don't process files in .git folder
+  // Don't process any files in a temp folder, unless the temp folder is used in inDir when running GoFSH
+  const tempDir = `${path.sep}temp${path.sep}`;
+  const gitDir = `${path.sep}.git${path.sep}`;
+  const tempInInDir = inDir.endsWith(`${path.sep}temp`) || inDir.includes(tempDir);
+  const relevantFiles = getFilesRecursive(inDir).filter(
+    file => (tempInInDir ? true : !file.includes(tempDir)) && !file.includes(gitDir)
+  );
```

Comment:
I feel like it might be cleaner (and more efficient) to do this in `getFilesRecursive`.  I'm thinking something like:
```ts
function getFilesRecursive(dir: string): string[] {
  if (fs.statSync(dir).isDirectory()) {
    const ancestors = fs.readdirSync(dir, 'utf8').map(f => {
      // NEW CODE STARTS HERE
      if (f === 'temp' || f === '.git') {
        logger.debug(`Skipping ${f} folder: ${path.join(dir, f)}`);
        return [];
      }
      // NEW CODE STOPS HERE
      return getFilesRecursive(path.join(dir, f))
    });
    return [].concat(...ancestors);
  } else {
    return [dir];
  }
}
```
Since it only looks at folder names we recursively find, it still allows for `temp` in the base folder passed in -- but would filter out any nested `temp` paths underneath the base (which I think is OK).

Reply:
The other advantage of putting this in `getFilesRecursive` is that we'd have more consistency w/ Fishing Trip.  The comment you brought up [here](https://github.com/FHIR/GoFSH/pull/58#discussion_r546955617) would be moot if the filter were inside `getFilesRecursive`.

Reply:
I like this idea. I was a little hesitant of making changes in `getFilesRecursive` because I wasn't sure if these were exceptions we wanted to make every time we need to get a list of files. But if you feel like they're common enough, and since it does solve the comment I made on Fishing Trip, I'm good with including them here.